### ISSUE TITLE
Reflect Interest back to staff/teachers

### ISSUE THREAD

**Issue by bethlakshmi** [bethlakshmi/GBE2#1098]
Depends on #1095 - this is a soup of ideas that need triage and definition by @burlexpo when we're ready to work on it:
- Create a report that teachers (and coordinator) can get to with the names of interested people for their classes.
- Make it possible for teachers (and class coordinator) to email students who show interest
- Have a bigger picture report of how many are interested in what class taught by what teacher available to class coordinators (includes Scratch)

**Comment by burlexpo** [comment]
@bethlakshmi can we kick this one up a notch? If it's not ready by Christmas, it's probably not super-useful for this year.  I think I might also like to get Mina's perspective.

**Comment by bethlakshmi** [comment]
OK, I'm planning to pick this up next...  (will coordinate push to production on that w. @burlexpo)... so now I'm putting serious design thought into this.

**1 - Teacher use case:**

Navigation:  Right now, teachers don't have a "special" menu.  My thought is that they get a link under any accepted class that says "n interested attendees" where "n" = the number of people who have expressed interest.  When the teacher clicks on the link, they go to an email page, that is a lot like the mail-to-bidders page, except it lets them choose from their collection of classes (checkboxes) and mail an email to the attendees.  The "from" is fixed to their email address.

We'll want a text blurb at the top explaining the feature.  Two key points:  (1) where this list of random people came from (they've shown interest in the class by adding it to their personal schedules, etc.).  (2) limitations of the system (namely, that it won't work with gmail/yahoo email addresses)

**2 - Teacher Error case**
I also want to mail the teacher an email if their email to interested users fails.  In other words, this case:
 1 - teacher finds our cool link
 2 - teacher uses it to email info to interested users (example, a dance teacher says to everyone to bring character shoes).
 3 - the teacher logs off happily to go pack for the expo.
 4 - the system tries to send email, and the email fails because the teacher has a gmail account and she didn't read our instructions.
 5 - the system mails an email *to* the teacher, reporting the problem.
 6 - now the teacher knows that people didn't get her mail, and she should either change her email address, or copy the list of email accounts and send it directly from her account.

That will necessitate another timed job, because the email is queued and then sent, so I can't discover the problem in process.  Given this fact, I think it's also useful to send email to ANY sender who's mail failed to send.  That way coordinators using the current functionality also get a heads up.

**3 - Coordinator case - emails**

For the Red/Scratch Coordinator case, I think we need a different flow.  Class by class isn't as interesting as the whole picture.

**Email case:**
1 - Coordinator goes to Special -> Contact -> Interested Attendees
2 - Coordinator can pick class bid state, b/c there may be a case of a class in "Withdrawn", "Duplicate" or even "Rejected" that accumulated some interest while it was previously in an accepted state.
   - Coordinator ALSO gets a list of all events that had any interest at all:
        - if the person is Class Coordinator - they get only classes.
        - if the person is a Schedule Maven - they get ALL events with any interest (that includes shows and special).  I'm willing to open it wider later (like to staff leads), but I'm trying to keep it a bit limited for now (this feature set is getting big).
3 - Coordinator can filter as desired.  If the coordinator picks a selected list of accepted classes AND does not include the Accepted state, the result is 0 interested people and an error warning.  (or else I do something clever with the UI that means that this outcome is unlikely).
4 - Mail flow works just like Mail to Bidders.

**4 - Coordinator interest awareness**

Also, there should be a non-email oriented report that lets us get info in the larger perspective... 

I really want to kill off the very ugly reports page:
http://gbeadmin.webfactional.com/reports

My proposal is to start it's death, by making a submenu for Special -> Reports -> Class Interest for this feature.   It'll be specifically about classes, because I want to think more about other types of interest before writing code about this.

Report: Interest by by class - a run down of every class, ordered by scheduled time.  Shows:
      - class name
      - teacher
      - location
      - max attendees (from bid form)
      - ?? style (ie, panel, workshop, movement, etc)??
      - # interested attendees

I'd do it w. the typical data table widget that lets the user sort and search and such.

I can think of other reports that might be spiffy (interest by teacher, for example, that sums up interest per teacher).  But I think that can wait.

@burlexpo - here's my proposal for detailed scope on this ticket.  It's grown quite a bit, and so I will probably want to do this in chunks.  My thought would be:

- Use case 3 - sm/med -  1 week
- Use case 1 & 2 - do first, medium size (1-2 week)
- Use case 4 - sm/med - 1 week, depending on the week.

That's a bet on priority order.   These are pretty independant, so I don't see any great savings from doing a particular order...

**Comment by bethlakshmi** [comment]
A very preliminary view of the teacher use case is now up on test:

Sample http://gbeadmin.webfactional.com/profile/landing_page/413

That's a way to preview a teacher who's class is accepted.  Or accept your own classes and play on your account.

**Comment by bethlakshmi** [comment]
@burlexpo  - this is the one that really wants feedback from you.
I updated the site today with the final draft for this:

http://gbeadmin.webfactional.com/gbe

shows teachers the interested people (for example, I just showed interest in YOUR class!)

special -> reports -> class interest 

shows the overall list of all accepted classes and interests.  

All blocks of text are editable in user messages.

Do I want to move the interest info to the center column? I could use a badge (in bootstrap terms) that looks like this:

https://getbootstrap.com/docs/3.3/components/#badges
